### ISSUE TITLE
Create simple 'tutorial' environment for drop-in education classes

### ISSUE THREAD

**Issue by planetf1** [odpi/egeria#1273]
In order to promote and educate interested developers on egeria, there is a need to support a simple 'drop in' lab approach where a student can quickly have access to the assets they need & work throgh a workbook in relation to a particular egeria oriented scenario - perhaps around patient record assets, or maybe server configuration.

One approach is to provide the individual pieces they need - the egeria code, kafka, zookeeper, and let them work through using the code, doc, rest api using curl or postman. 

However this is error prone - many moving parts to get in place before starting to learn egeria.

One possibility would be to use the helm charts we've already created. They are simple to install, but only if prereqs are met - access to a kubernetes environment, skills to use, security. We do benefit from re-use and high functionality

However kubernetes on a laptop can be problematic - for example the MacOS version currently results in significant CPU drain and battery impact - a problem for roaming users at a conference.

Creating a plain docker environment is attractive as it's a single environment to launch, but assembling a variety of components can be somewhat fragile

In between these two approaches is to use docker compose, or docker stacks -- this has far less capability than k8s, but is simple, and has much lower overhead in terms of requirements.

Further, rather than make use of docs in tools like Microsoft Word, we can use notebooks which include both text & python snippets to exercise the rest API. This too can be added to the combined environment

**Comment by planetf1** [comment]
cc: @mandy-chessell

**Comment by planetf1** [comment]
Current proposal is
 * Use docker-compose
 * Include jupyter notebook (lab) for supporting reading of text, launching python code against egeria
 * Include egeria, kafka, zookeeper

One created I will contact users to test - across Windows, Linux & MacOS

**Comment by planetf1** [comment]
First version added to fork above
 - launches zookeeper, kafka, egeria, jupyter notebook lab
 - kafka, zookeeper use non standard ports to avoid clashes (+10000)
 
Still to do
 - pom
 - README
 - Agree final directory location
 - Decide way to add actual notebook content
 - Preconfiguration of egeria configuration

**Comment by planetf1** [comment]
Added simple readme

**Comment by planetf1** [comment]
If anyone would like to try this, start up the environment, but when using egeria, access it via http://egeria:8080 (not localhost), and for kafka use kafka:9082

With the notebook started you can click the ^ (up arrow) on the left hand side to upload a notebook file if you have one, or can create a new one.

Bear in mind that egeria comes up but without preconfiguration so you'll need to walk through the usual steps in the python notebook. Perfect for teaching admin ... though I'll look at flexible ways to get other assets preloaded as needed (notebooks or configuration)

**Comment by planetf1** [comment]
I've posted my initial prototype notebook to https://gist.githubusercontent.com/planetf1/24b7c06dcb6c40b30a145f0b431200d3/raw/5c0fc3f324df2fc5d83ae4691970b82e86aab22c/EgeriaTutorial.ipynb

This *will not* work directly with the image here for two reasons
 1. you'll need to change the ports & hostname
 2 you'll need to initialize the egeria server (cohort etc)

but is provided as a starting template.

preloading of an egeria config is not in place.

**Comment by planetf1** [comment]
See the following file for an example of configuring the server

This also proves basic connectivity though I've not yet checked kafka

https://gist.githubusercontent.com/planetf1/2e2b0a468bc0d32e3bf87d5b84d1a499/raw/fb46dcd3b12bf7e8845f7e7d8ff3a7d3a6ea879c/EgeriaAdmin.ipynb

**Comment by planetf1** [comment]
I was able to run the image on windows in exactly the same way. NO changes, the notebook worked just fine, as did the sample Admin tutorial.

I didn't have docker installed, though I did previously, and Hyper-V (which docker requires) was enabled.

I did note that the download was large (though quick on my connection) with the expanded images after running the composition:
```
nigel@MEGA C:\Users\nigel
$ docker images
REPOSITORY                 TAG                 IMAGE ID            CREATED             SIZE
odpi/egeria                latest              5ceaeb9e0a6b        6 hours ago         208MB
bitnami/kafka              latest              a31b189faac7        9 hours ago         335MB
bitnami/zookeeper          latest              d40cd3e96ddf        14 hours ago        270MB
jupyter/minimal-notebook   latest              7233a81c81d8        9 days ago          2.67GB
```

In particular the jupyter notebook image is large
 (though this is after being unpacked)

As a comparison, on my MacOS 'homebrew' install of jupyter, the size is 67 MB

**Comment by planetf1** [comment]
See https://github.com/jupyter/docker-stacks/issues/775

**Comment by planetf1** [comment]
Switched to lighter jupyter image, sufficient for demoing egeria - thanks for tip @CDaRip2U

**Comment by planetf1** [comment]
Added 2 additional egeria services - and renamed original
So we now have
 - egeriacore
 - egeriadev
 - egeriadl

Docker compose does not support 'replicas' (unlike docker swarm, or effectively k8s) so the entire entry is replicated. switching to docker swarm from compose adds more complexity  which needs more evaluation

Note any existing notebooks need to be updated to use the new server names

**Comment by planetf1** [comment]
Note that due to the way each container has it's own hostnames, the egeria servers are now at
 http://egeriacore:8080 (also aliases to http://egeria:8080 for compatability with previous notebooks)
 http://egeriadev:8080
 http://eegeriadl:8080

as accessed from the notebook. The README.md has further info
cc: @mandy-chessell

**Comment by planetf1** [comment]
any additional needs/bugs should be raised in new issue
